react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -1,38 +1,43 @@
+/* eslint-disable no-plusplus */
+/* eslint-disable no-console */
 import React, { Component, Fragment } from 'react';
-import store from '@redux/store';
+import { connect } from 'react-redux';
 import Navbar from '@components/Navbar';
 import Footer from '@components/Footer';
+import { func, arrayOf } from 'prop-types';
+import { bookPropType } from '@constants/propTypes';
+
+import actionsCreators from '../../../redux/book/actions';
 
 import Book from './components/Book';
 import Search from './components/Search';
 import ShoppingCart from './components/ShoppingCart';
 import styles from './styles.scss';
 
 class App extends Component {
-  state = {
-    books: [],
-    bookSelected: []
-  };
+  componentDidMount = () => this.props.getBooks();
 
-  componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
-  }
-
-  // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    const { originalData, searchBooks } = this.props;
+    const books = originalData.filter(book => book.name.toLowerCase().includes(value));
+    searchBooks(books);
+  };
 
-  // TODO to implement the dispatch
-  addToCart = item => {};
+  addToCart = item => this.props.addToCart(item);
 
-  // TODO to implement the dispatch
-  addItem = itemId => {};
+  addItem = itemId => {
+    const { addItem, bookSelected } = this.props;
+    const books = [...bookSelected];
+    const book = books.find(({ id }) => id === itemId);
+    book.quantity++;
+    addItem(books);
```

Comment:
I think there is a mistake here 👀 you are modifying book quantity key and then adding bookS instead of book in the `addItem()` method.

Also, if you are not touching bookSelected array or modifying it in anyway, then you can use it instead of making a copy of it.